python/pyarrow/_parquet.pyx

```diff
@@ -952,6 +953,21 @@ cdef class ParquetReader(_Weakrefable):
 
         arrow_props.set_pre_buffer(pre_buffer)
 
+        if (
+            coerce_int96_timestamp_unit is None or
+            coerce_int96_timestamp_unit == "ns"
+        ):
+            arrow_props.set_coerce_int96_timestamp_unit(TimeUnit_NANO)
+        elif coerce_int96_timestamp_unit == "us":
+            arrow_props.set_coerce_int96_timestamp_unit(TimeUnit_MICRO)
+        elif coerce_int96_timestamp_unit == "ms":
+            arrow_props.set_coerce_int96_timestamp_unit(TimeUnit_MILLI)
+        elif coerce_int96_timestamp_unit == "s":
+            arrow_props.set_coerce_int96_timestamp_unit(TimeUnit_SECOND)
+        else:
```

Comment:
How about:
```
elif coerce_int96_timestamp_unit is not None:
```

so we'd leave `coerce_int96_timestamp_unit` to be set by the C++ implementation by default?